pkg/common/common_test.go

```diff
@@ -67,3 +71,192 @@ func TestFind(t *testing.T) {
+		}
+	})
+	t.Run(" when a non-reserved env var name is used as the key then return the default value", func(t *testing.T) {
+		key := "MY_VAR"
+		defaultVal := ""
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != defaultVal {
+			t.Errorf("Expected %v, but got %v", defaultVal, result)
+		}
+	})
+}
+
+func TestGetDefaultIfNil(t *testing.T) {
+	t.Run("when value is non-nil pointer type and default value is provided then return value", func(t *testing.T) {
+		val := "test"
+		def := "default"
+
+		result := GetDefaultIfNil(&val, def)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
+	t.Run("when value is nil pointer type and default value is provided then return default value", func(t *testing.T) {
+		var val *string
+		def := "default"
+
+		result := GetDefaultIfNil(val, def)
+
+		if result != def {
+			t.Errorf("Expected %v, but got %v", def, result)
+		}
+	})
+	t.Run("when value is not a pointer type and default value is provided then return default value", func(t *testing.T) {
+		var val int
+		def := 123
+
+		result := GetDefaultIfNil(&val, def)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
+}
+
+func TestGetEmptySliceIfNil(t *testing.T) {
+	t.Run("when a non-nil slice is provided then return same slice", func(t *testing.T) {
+		value := []int{1, 2, 3}
+		expected := value
+
+		result := GetEmptySliceIfNil(value)
+
+		if !reflect.DeepEqual(result, expected) {
+			t.Errorf("Expected %v, but got %v", expected, result)
+		}
+	})
+
+	t.Run("when a nil slice is provided then return an empty slice of the same type", func(t *testing.T) {
+		var value []int
+		expected := []int{}
+
+		result := GetEmptySliceIfNil(value)
+
+		if !reflect.DeepEqual(result, expected) {
+			t.Errorf("Expected %v, but got %v", expected, result)
+		}
+	})
+}
+
+func TestNamespacedNameToObjectKey(t *testing.T) {
+	t.Run("when a namespaced name is provided then return an ObjectKey with corresponding namespace and name", func(t *testing.T) {
+		namespacedName := "test-namespace/test-name"
+		defaultNamespace := "default"
+
+		result := NamespacedNameToObjectKey(namespacedName, defaultNamespace)
+
+		expected := client.ObjectKey{Name: "test-name", Namespace: "test-namespace"}
+		if !reflect.DeepEqual(result, expected) {
+			t.Errorf("Expected %v, but got %v", expected, result)
+		}
+	})
+
+	t.Run("when only a name is provided then return an ObjectKey with the default namespace and provided name", func(t *testing.T) {
+		namespacedName := "test-name"
+		defaultNamespace := "default"
+
+		result := NamespacedNameToObjectKey(namespacedName, defaultNamespace)
+
+		expected := client.ObjectKey{Name: "test-name", Namespace: "default"}
+		if !reflect.DeepEqual(result, expected) {
+			t.Errorf("Expected %v, but got %v", expected, result)
+		}
+	})
+
+	t.Run("when an empty string is provided, then return an ObjectKey with default namespace and empty name", func(t *testing.T) {
```

Comment:
I'm wondering if we should also return error, since it's mandatory to have a _name_ for querying k8s API... wdyt @eguzki ?

Reply:
having `NamespacedNameToObjectKey` return an error for the case the name is empty seems overcomplicating. The function does what it says and if you provide an empty string, you would get empty name (plus the default namespace). I think that the error triggering should lay on the callers, not on this method. But just my opinion...